homeassistant/components/fmi/weather.py

```diff
@@ -0,0 +1,183 @@
+"""Support for retrieving meteorological data from FMI (Finnish Meteorological Institute)."""
+from dateutil import tz
+
+from homeassistant.components.weather import (
+    ATTR_FORECAST_CONDITION,
+    ATTR_FORECAST_PRECIPITATION,
+    ATTR_FORECAST_TEMP,
+    ATTR_FORECAST_TIME,
+    ATTR_FORECAST_WIND_BEARING,
+    ATTR_FORECAST_WIND_SPEED,
+    ATTR_WEATHER_HUMIDITY,
+    ATTR_WEATHER_PRESSURE,
+    WeatherEntity,
+)
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import get_weather_symbol
+from .const import _LOGGER, ATTRIBUTION, COORDINATOR, DOMAIN, MANUFACTURER, NAME
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add an FMI weather entity from a config_entry."""
+    name = config_entry.data[CONF_NAME]
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
+
+    async_add_entities([FMIWeatherEntity(name, coordinator)], False)
+
+
+class FMIWeatherEntity(CoordinatorEntity, WeatherEntity):
+    """Define an FMI Weather Entity."""
+
+    def __init__(self, name, coordinator):
+        """Initialize FMI weather object."""
+        super().__init__(coordinator)
+        self._name = name
+        self._attrs = {}
+        self._unit_system = "Metric"
+        self._fmi = coordinator
+
+    @property
+    def name(self):
+        """Return the name of the place based on Lat/Long."""
+        if self._fmi is None:
+            return self._name
+
+        if self._fmi.current is None:
+            return self._name
+
+        return self._fmi.current.place
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def unique_id(self):
+        """Return a unique_id for this entity."""
+        return self.coordinator.unique_id
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self.coordinator.unique_id)},
+            "name": NAME,
+            "manufacturer": MANUFACTURER,
+            "entry_type": "service",
+        }
+
+    @property
+    def available(self):
+        """Return if weather data is available from FMI."""
+        if self._fmi is None:
+            return False
+
+        return self._fmi.current is not None
```

Comment:
```suggestion
        return self.coordinator.current is not None
```

(And so on for all other occurrences in this PR)